### PR TITLE
Router: add a Router class

### PR DESCRIPTION
**Pull request by SansGodasse**
This branch also adds a Dependancies Injector Container and a Route class working together to execute controller methods.

### CODE REVIEW

File: src/Router/Router.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                isset($_GET['post-id']) &&
+                is_numeric($_GET['post-id'])) {
+
+                $controller = 'Controller\BlogController';
+                $method = 'showASinglePost';
+                $params = ['postId' => $_GET['post-id']];
+
+            // Blog Admin
+            } elseif ($page === 'blog-admin') {
+                if (isset($_POST['add-post'])) {
```
Comment (olix21) [inline src/Router/Router.php]: Why are you using $_POST for all your admin routing? Why not something like /admin/post?action=add_post
Or
/admin/post/add?

File: src/Router/Router.php [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    
+    /**
+     * Analyze the url and return the controller name, the method to call and the parameters
+     *
+     * @return Route
+     */
+    public static function run(): Route
+    {
+        // Default route : Home
+        $controller = 'Controller\BlogController';
```
Comment (olix21) [inline src/Router/Router.php]: `BlogController::class` ou en tout cas échapper l'antislash

**Comment by SansGodasse** [comment]
[HELP] I now use urls like `admin/post-editor` instead of `?page=post-editor` but it creates bugs like css file not found or strange links like admin/admin...

What should I do to correct this?

**Comment by olix21** [comment]
> [HELP] I now use urls like `admin/post-editor` instead of `?page=post-editor` but it creates bugs like css file not found or strange links like admin/admin...
> 
> What should I do to correct this?

Be sure to use absolute urls and not relative ones (add `/` to make sure you're using absolute urls)

**Comment by SansGodasse** [comment]
> 
> 
> > [HELP] I now use urls like `admin/post-editor` instead of `?page=post-editor` but it creates bugs like css file not found or strange links like admin/admin...
> > What should I do to correct this?
> 
> Be sure to use absolute urls and not relative ones (add `/` to make sure you're using absolute urls)

I tried href="/admin" but I get localhost/admin in the navigator.

**Comment by olix21** [comment]
it's because of your server configuration. You have a main entry point (localhost) where you can find all your website...

What you can do is starting the php builtin server where your index.php is.
To start the builtin server: `php -S localhost:8000` from your `project_folder/public` or `php -S localhost:8000 -t public` from your project folder

Then, you can go to localhost:8000 and access to your project. If you don't have php installed by default, you can use the one from Mamp/Xamp/Wamp or whatever server you're using. Juste replace `php` with the full path to the M.X.W.amp php

**Comment by SansGodasse** [comment]
Ok, I launched a server from the public folder and it works... But now the errors messages are awful and $_GET['url'] doesn't exist anymore...

What can I do to have nice errors and $_GET['url'] ?

**Comment by olix21** [comment]
check the $_REQUEST. What do you mean by "awful"? :p

**Comment by SansGodasse** [comment]
`$_REQUEST` only contains parameters and no url.

**Comment by SansGodasse** [comment]
Ok, the branch is ready to merge. What do you think ?

**Comment by olix21** [comment]
merge master in this branch, there are still conflicts with the target branch

**Comment by SansGodasse** [comment]
Ok, I didn't know I should've done that. Now it's done. Can I merge?

**Pull request merged**